### PR TITLE
add utorid as a part of the hash so its a different for everyone

### PR DESCRIPTION
**Pull request by logonoff**

### CODE REVIEW

File: backend/src/common/tierlist.ts [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
 
+/** @return a two letter hash of the string */
+export const twoLetterHash = (str: string) => {
+    let hash = 0;
+    for (let i = 0; i < str.length; i++) {
+        hash = ((hash << 5) - hash) + str.charCodeAt(i);
+        hash |= 0;
+    }
+    return Math.abs(hash).toString(36).substr(0, 2);
+};
```
Comment (ggggg) [inline backend/src/common/tierlist.ts:17]: Im assuming you tested this and it worked

File: backend/src/common/tierlist.ts [hunk 2]
```diff
@@ -24,8 +47,8 @@ function generateList(assignment: Omit<FullFetchedAssignment, "due_date">, user?
     }
     const scores = assignment.submissions.map(submission =>
         ({
-            you: submission.author.utorid === (user ? (typeof user === "string" ? user : user.utorid) : false),
-            name: submission.author.email[0] + submission.author.email[submission.author.email.indexOf(".") + 1],
+            you: user ? isSelf(user, submission.author.utorid) : false,
+            name: (user ? isSelf(user, submission.author.utorid) : false) ? getUserInitials(submission.author) : twoLetterHash(submission.author.utorid + (user ? getUtorid(user) : "")),
```
Comment (ggggg) [inline backend/src/common/tierlist.ts:51]: kinda messy but its fine

**Review by ggggg — APPROVED** [review]

File: backend/src/common/tierlist.ts (hunk 1 above)
Comment (logonoff) [reply backend/src/common/tierlist.ts:17]: yes

File: backend/src/common/tierlist.ts (hunk 2 above)
Comment (logonoff) [reply backend/src/common/tierlist.ts:51]: yeah idk how else to do it

**Pull request merged**